x-pack/solutions/observability/plugins/infra/test/scout/ui/fixtures/page_objects/inventory.ts

```diff
@@ -5,13 +5,173 @@
+  public readonly inventorySwitcherContainersButton: Locator;
+
+  public readonly k8sTourText: Locator;
+  public readonly k8sTourDismissButton: Locator;
+
+  public readonly mapViewButton: Locator;
+  public readonly waffleMap: Locator;
+  public readonly toggleTimelineButton: Locator;
+  public readonly timelineContainerClosed: Locator;
+  public readonly timelineContainerOpen: Locator;
+
+  public readonly tableViewButton: Locator;
+  public readonly nodesOverviewTable: Locator;
+
+  public readonly noDataPrompt: Locator;
+
+  public readonly noDataPage: Locator;
+  public readonly noDataPageActionButton: Locator;
+
+  constructor(private readonly page: ScoutPage, private readonly kbnUrl: KibanaUrl) {
+    this.feedbackLink = this.page.getByTestId('infraInventoryFeedbackLink');
+    this.k8sFeedbackLink = this.page.getByTestId('infra-kubernetes-feedback-link');
+
+    this.datePickerInput = this.page.getByTestId('waffleDatePicker').getByRole('textbox');
+
+    this.inventorySwitcherButton = this.page.getByTestId('openInventorySwitcher');
+    this.inventorySwitcherHostsButton = this.page.getByTestId('goToHost');
+    this.inventorySwitcherPodsButton = this.page.getByTestId('goToPods');
+    this.inventorySwitcherContainersButton = this.page.getByTestId('goToContainer');
+
+    this.k8sTourText = this.page.getByTestId('infra-kubernetesTour-text');
+    this.k8sTourDismissButton = this.page.getByTestId('infra-kubernetesTour-dismiss');
+
+    this.mapViewButton = this.page.getByRole('button', { name: 'Map view' });
+    this.waffleMap = this.page.getByTestId('waffleMap');
+    this.toggleTimelineButton = this.page.getByTestId('toggleTimelineButton');
+    this.timelineContainerClosed = this.page.getByTestId('timelineContainerClosed');
+    this.timelineContainerOpen = this.page.getByTestId('timelineContainerOpen');
+
+    this.tableViewButton = this.page.getByRole('button', { name: 'Table view' });
+    this.nodesOverviewTable = this.page.getByTestId('infraNodesOverviewTable');
+
+    this.noDataPrompt = this.page.getByTestId('noMetricsDataPrompt');
+
+    this.noDataPage = this.page.getByTestId('kbnNoDataPage');
+    this.noDataPageActionButton = this.noDataPage.getByTestId('noDataDefaultActionButton');
+  }
+
+  private async startWaitingForSnapshotRequest() {
+    // Once the page has loaded once, visual loading indicator doesn't always appear on subsequent loads.
+    // So we need to wait for the network request too to ensure new data has been retrieved.
+    return this.page.waitForRequest((request) => request.url().includes('/api/metrics/snapshot'));
+  }
+
+  private async waitForNodesToLoad(snapshotPromise?: Promise<Request>) {
+    const panelPromise = this.page
+      .getByTestId('infraNodesOverviewLoadingPanel')
+      .waitFor({ state: 'hidden' });
+
+    if (snapshotPromise) {
+      await Promise.all([panelPromise, snapshotPromise]);
+    } else {
+      await panelPromise;
+    }
+  }
```

Comment:
we recently saw a failure for the similar case: I suspect Playwright can't properly handle exceptions coming from `Promise.all` and you won't get a stacktrace to understand what condition caused the failure.

If API call completion matters, I would first wait for the response and (optionally) status code:

```ts
// Wait for successful API completion
await this.page.waitForResponse(resp => 
  resp.url().includes('/api/metrics/snapshot') && resp.status() === 200
);
```

Then you can check if particular component/element is displayed and proceed with the test flow.